en/docs/deploy-and-publish/deploy-on-gateway/api-microgateway/deploy-api/deploy-api-to-microgateway.md

```diff
@@ -0,0 +1,146 @@
+
+2. [Via the CLI tool (apictl)](#via-the-cli-tool-apictl)
+
+    If you are not using WSO2 API Manager yet, or only want to tryout microgateway, simply use the CLI tool [apictl](https://github.com/wso2/product-apim-tooling/releases) to host your API.
+
+## Via API Manager
+
+!!! info
+    **Before you begin**
+
+    - Make sure you have installed Docker and Docker Compose on your machine.
+
+    - Download the latest [Microgateway 4.0.0 release](https://github.com/wso2/product-microgateway/releases)and extract it to a folder of your choice. The extracted folder will be referred to as `MG_HOME` here onwards.
+
+    - This guide assumes that you have already started the WSO2 API Manager instance. If not, download the latest [release](https://github.com/wso2/product-apim/releases)and follow the steps [here](https://github.com/wso2/product-apim#installation--running)
+
+### Step 1 - Find the APIM IP Address
+
+In order to tell Microgateway where API Manager (APIM) is located, find out the IP that can be used to access the API Manager instance. If you are trying out WSO2 API Manager locally, the private IP retrived using `hostname -I` or `ipconfig` would do.
+
+### Step 2 - Update the Microgateway Config File
+
+Open the `<MGW_HOME>/resources/conf/config.toml` file in a text editor and update it as follows.
+
+In the `[controlPlane.eventHub]` section,
+
+ - set `enabled` to true
+ - update `serviceUrl` with the IP and the port of API Manager
+ - if you want to use a Gateway Environment other than the default, update `environmentLabels` with the name of the new Gateway Environment. If not, leave the default value `"Production and Sandbox"` as it is.
+
+ Example
+ ``` yaml
+ [controlPlane.eventHub]
+  enabled = true
+  serviceUrl = "https://192.168.1.94:9443/"
+  username="admin"
+  password="$env{cp_admin_pwd}"
+  environmentLabels = ["Production and Sandbox"]
+  retryInterval = 5
+  skipSSLVerification=true
+  # Message broker connection URL of the control plane
+  [controlPlane.eventHub.jmsConnectionParameters]
+    eventListeningEndpoints = ["amqp://admin:$env{cp_admin_pwd}@apim:5672?retries='10'&connectdelay='30'"]
+ ``` 
+
+!!! tip
+
+    In API Manager, a new Gateway Environment can be created from the Admin Portal (available at `https:<apim-host>:<apim-port>/admin`) **Gateways** tab.
+
+### Step 3 - Start the Microgateway
+
+Now, let's start the microgateway. Navigate to `MG_HOME` and execute the following command.
+    
+``` bash
+docker-compose up -d
+```
+
+That's it!
+
+
+During the startup, Microgateway will check the `config.toml` to see if the `controlPlane.eventHub` configuration has been enabled. If so, it will start fetching all the necessary artifacts that belongs to the gateway environment given in `environmentLabels`. These artifacts include deployed APIs, Applications, Subscriptions, Polices, information related to Key Managers, and etc.
```

Comment:
```suggestion
During the startup, Microgateway will check the `config.toml` to see if the `controlPlane.eventHub` configuration has been enabled. If so, it will start fetching all the necessary artifacts that belongs to the gateway environment given in `environmentLabels`. These artifacts include deployed APIs, Applications, Subscriptions, Polices, information related to Key Managers, etc.
```

Reply:
Added the change